google/resource_storage_bucket.go

```diff
@@ -382,3 +453,130 @@ func flattenCors(corsRules []*storage.BucketCors) []map[string]interface{} {
 	}
 	return corsRulesSchema
 }
+
+func resourceGCSBucketLifecycleCreateOrUpdate(d *schema.ResourceData, sb *storage.Bucket) error {
+	if v, ok := d.GetOk("lifecycle_rule"); ok {
+		lifecycle_rules := v.([]interface{})
+
+		len_lifecycle_rules := len(lifecycle_rules)
+		if len_lifecycle_rules > 100 {
+			return fmt.Errorf("At most 100 lifecycle_rule blocks are allowed, %d given", len_lifecycle_rules)
+		}
+
+		sb.Lifecycle = &storage.BucketLifecycle{}
+		sb.Lifecycle.Rule = make([]*storage.BucketLifecycleRule, 0, len_lifecycle_rules)
+
+		for _, lifecycle_rule := range lifecycle_rules {
+			lifecycle_rule := lifecycle_rule.(map[string]interface{})
+
+			target_lifecycle_rule := &storage.BucketLifecycleRule{}
+
+			if v, ok := lifecycle_rule["action"]; ok {
+				action := v.(*schema.Set).List()[0].(map[string]interface{})
```

Comment:
I apologise, you were correct--MaxItems and MinItems would have handled that constraint for you. But the extra check also doesn't hurt.